tree: refactor FmtCtx to use option pattern

otan opened the pull request:
There are many setters or optional arguments to FmtCtx. Consolidate this
altogether by using the option pattern.

Refs: #65724

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/66878)
<!-- Reviewable:end -->

otan committed:
tree: refactor FmtCtx to use option pattern

There are many setters or optional arguments to FmtCtx. Consolidate this
altogether by using the option pattern.

Release note: None

rafiss reviewed (approved):
nice!

otan commented:
bors r=rafiss
